server/backend/api/app/routers/router_guards.py

```diff
@@ -30,25 +40,68 @@
 ERROR_EXTERNAL_USER_ACTION_PROHIBITED = "external_user_action_prohibited"
 
 no_requester_exception = HTTPException(
-    status_code=HTTPStatus.FORBIDDEN, # 403
+    status_code=HTTPStatus.FORBIDDEN,  # 403
     detail={
         "code": ERROR_REQUESTER_NOT_EXISTS,
         "description": "Requester does not exist, action is not allowed",
-    }
+    },
 )
 
 external_user_prohibited_exception = HTTPException(
     status_code=HTTPStatus.FORBIDDEN,
     detail={
         "code": ERROR_EXTERNAL_USER_ACTION_PROHIBITED,
         "description": "Action is not allowed for external user.",
-    }
+    },
 )
 
+
+async def get_current_requester(
+    request_cognito_user_id: str = Depends(get_request_cognito_user_id),
+    db: Session = Depends(database.get_db),
+):
+    fam_user: FamUser = crud_user.get_user_by_cognito_user_id(
+        db, request_cognito_user_id
+    )
+    if fam_user is None:
+        raise no_requester_exception
+
+    requester = Requester.model_validate(fam_user)
+    LOGGER.debug(f"Current request user (requester): {requester}")
+    return requester
+
+
+def authorize(
+    claims: dict = Depends(validate_token),
+    db: Session = Depends(database.get_db),
+    requester: Requester = Depends(get_current_requester),
+):
+    # we require user to be the app admin or delegated admin of at least one application
+    if JWT_GROUPS_KEY not in claims or len(claims[JWT_GROUPS_KEY]) == 0:
+        # if user has no application admin access
+        # check if user has any delegated admin access
+        user_access_control_privilege = (
+            crud_access_control_privilege.get_delegated_admin_by_user_id(
+                db, requester.user_id
+            )
+        )
+        # if user is not app admin and not delegated admin of any application, throw miss access group error
+        if len(user_access_control_privilege) == 0:
+            raise HTTPException(
+                status_code=403,
+                detail={
+                    "code": ERROR_GROUPS_REQUIRED,
+                    "description": "At least one access group is required",
+                },
+                headers={"WWW-Authenticate": "Bearer"},
+            )
+
+
 def authorize_by_app_id(
     application_id: int,
     db: Session = Depends(database.get_db),
-    claims: dict = Depends(validate_token)
+    claims: dict = Depends(validate_token),
+    requester: Requester = Depends(get_current_requester),
```

Comment:
this authorize_by_app_id method will check is user is the admin or delegated admin of the specific application

Reply:
Again, I think this should be a comment on the method.

Reply:
Or name the methods more specifically

Reply:
yeah, I added comment as well